src/types.py

```diff
@@ -311,7 +315,9 @@ def simple_ctype(typename: str) -> ca.TypeDecl:
         size = data.size or 32
         sign = "s" if data.sign & TypeData.SIGNED else "u"
 
-        if (data.kind & TypeData.K_ANYREG) == TypeData.K_ANYREG:
+        if (
+            data.kind & TypeData.K_ANYREG
+        ) == TypeData.K_ANYREG and data.likely_kind & TypeData.K_FLOAT:
```

Comment:
This was the one part that seems a bit too specific? But I see you change it up in `likely-floats`, so I think it's fine for now.